omas/omas_plot.py

```diff
@@ -446,3 +514,473 @@ def core_profiles_pressures(ods, time_index=0, ax=None, **kw):
+        ax.text(r, z, label, color=gas_mark[0].get_color(),
+                va=['top', 'bottom'][int(z > 0)], ha=['left', 'right'][int(r < rsplit)])
+    return
+
+
+@add_to__ODS__
+def pf_active_overlay(ods, ax=None, **kw):
+    """
+    Plots overlays of interferometer chords.
+
+    :param ods: OMAS ODS instance
+
+    :param ax: Axes instance
+
+    :param \**kw: Additional keywords
+        scalex, scaley: passed to ax.autoscale_view() call at the end
+
+        * Remaining keywords are passed to matplotlib.patches.Polygon call
+            Hint: you may want to set facecolor instead of just color
+    """
+    # Make sure there is something to plot or else just give up and return
+    nc = get_channel_count(
+        ods, 'pf_active', check_loc='pf_active.coil.0.element.0.geometry.oblique.r', channels_name='coil',
+        test_checker='checker > 0')
+    if nc == 0:
+        return
+
+    if ax is None:
+        ax = pyplot.gca()
+
+    kw.setdefault('label', 'Active PF coils')
+    kw.setdefault('facecolor', 'gray')
+    kw.setdefault('edgecolor', 'k')
+    kw.setdefault('alpha', 0.7)
+    scalex, scaley = kw.pop('scalex', True), kw.pop('scaley', True)
+
+    patches = []
+    for i in range(nc):  # From  iris:/fusion/usc/src/idl/efitview/diagnoses/DIII-D/coils.pro ,  2018 June 08  D. Eldon
+        oblique = ods['pf_active.coil'][i]['element.0.geometry.oblique']
+        fdat = [oblique['r'], oblique['z'], oblique['length'], oblique['thickness'], oblique['alpha'], oblique['beta']]
+
+        xarr = [
+            fdat[0] - fdat[2] / 2. - fdat[3] / 2. * numpy.tan(numpy.pi/2. - fdat[5]),
+            fdat[0] - fdat[2] / 2. + fdat[3] / 2. * numpy.tan(numpy.pi/2. - fdat[5]),
+            fdat[0] + fdat[2] / 2. + fdat[3] / 2. * numpy.tan(numpy.pi/2. - fdat[5]),
+            fdat[0] + fdat[2] / 2. - fdat[3] / 2. * numpy.tan(numpy.pi/2. - fdat[5]),
+        ]
+        yarr = [
+            fdat[1] - fdat[3] / 2. - fdat[2] / 2. * numpy.tan(fdat[4]),
+            fdat[1] + fdat[3] / 2. - fdat[2] / 2. * numpy.tan(fdat[4]),
+            fdat[1] + fdat[3] / 2. + fdat[2] / 2. * numpy.tan(fdat[4]),
+            fdat[1] - fdat[3] / 2. + fdat[2] / 2. * numpy.tan(fdat[4]),
+        ]
+        path = numpy.array([xarr, yarr]).T
+        patches.append(matplotlib.patches.Polygon(path, closed=True, **kw))
+        kw.pop('label', None)  # Prevent label from being placed on more than one patch
+
+    for p in patches:
+        ax.add_patch(p)  # Using patch collection breaks auto legend labeling, so add patches individually.
+
+    ax.autoscale_view(scalex=scalex, scaley=scaley)  # add_patch doesn't include this
+
+    return
+
+
+@add_to__ODS__
+def magnetics_overlay(ods, ax=None, **kw):
+    """
+    Plots overlays of magnetics: B_pol probes and flux loops
+
+    :param ods: OMAS ODS instance
+
+    :param ax: Axes instance
+
+    :param \**kw: Additional keywords
+        show_bpol_probe: bool
```

Comment:
@eldond I feel that some of these arguments should be explicit and not be in `**kw`. For the hardware specific overlay function (not the general one) I would leave in `**kw` only the plotting options that are passed to matplotlib.

Reply:
Okay, I made most of them explicit. There were a few places where it felt more natural to keep them inside **kw, but I can change this if you like. Still in **kw:
 - general keywords that all the hardware overlays accept (`labelevery` and `mask`)
 - CER overlay colors and markers. I loop through strings and pop them. It seems nicer that way. I guess I could `eval` or loop through a `zip` if I made these keywords explicit.
 - scalex and scaley in `pf_active_overlay`: these are popped off and passed to `autoscale_view`. The user probably won't bother with them, so I want to try to tuck them away. The matplotlib function I grabbed this idea from did it this way.